airflow/typing_compat.py

```diff
@@ -31,8 +31,14 @@
 from typing import Protocol, TypedDict, runtime_checkable
 
 # Literal from typing module has various issues in different Python versions, see:
-# https://typing-extensions.readthedocs.io/en/latest/#Literal
-from typing_extensions import Literal
+# - https://typing-extensions.readthedocs.io/en/latest/#Literal
+# - bpo-45679: https://github.com/python/cpython/pull/29334
+# - bpo-42345: https://github.com/python/cpython/pull/23294
+# - bpo-42345: https://github.com/python/cpython/pull/23383
+if sys.version_info >= (3, 10, 1) or (3, 9, 8) <= sys.version_info < (3, 10):
+    from typing import Literal
+else:
+    from typing_extensions import Literal  # type: ignore[assignment]
```

Comment:
Without `type: ignore[assignment]` mypy complains about type assigments

```console
airflow/typing_compat.py:41: error: Incompatible import of "Literal" (imported
name has type "typing_extensions._SpecialForm", local name has type
"<typing special form>")  [assignment]
        from typing_extensions import Literal
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Found 1 error in 1 file (checked 1 source file)
```

Another fix: set condition to `if sys.version_info >= (3, 11):`